quiche/src/lib.rs

```diff
@@ -3177,11 +3177,7 @@ impl Connection {
             }
         }
 
-        let mut left = b.cap();
-
-        // Limit output packet size by congestion window size.
-        left =
-            cmp::min(left, self.paths.get(send_pid)?.recovery.cwnd_available());
+        let mut buffer_capacity = b.cap();
```

Comment:
I think it would be cleaner to keep the `left` name, and simply move the `left = cmp::min(left, cwnd_available);` line to after the ACK frame is created? This way it seems the change would be smaller, and we wouldn't need the additional `consumed_by_acks` logic.

Reply:
Thanks for the review !
I still have a small question, if we put:

```
left = cmp::min(left, self.paths.get(send_pid)?.recovery.cwnd_available())
```

after the ACKs have been packed, how would we prevent the resulting packet to be larger than `cwnd_available` without using the `consumed_by_acks` logic ? We still need to subtract the bytes of the ACK frame from the available cwnd at some point to avoid packing too large packets. Or do we consider it is not that big of a deal ?

Reply:
I'm not sure I understand the question. `push_frame_to_pkt!()` will decrease `left` by the amount of ACK frame bytes written, so doing `cmp::min(left, cwnd_available)` afterwards will already take the ACK frame size into consideration? And if `cwnd_available` is zero then `left` becomes zero too. Am I missing something?

Reply:
I was just thinking of the case where `cwnd_available` is both smaller than `b.cap()` and larger than 0. 

For instance if `b.cap()` is 1000, `cwnd_available` is 500 and we want to bundle an ACK frame of 10 bytes.
If I understand everything correctly, we start with `left` equal to `b.cap()` (=1000). We push the 10 bytes ACK frame into the packet so `left` is decreased by 10 and is now equal to 990. Then, we do `left = cmp::min(left, cwnd_available)`, so it becomes 500 because `cwnd_available` is equal to 500.

So it means that we can pack 500 more bytes of e.g. stream frames, resulting to a payload of 510 bytes (10 bytes for the ACK and 500 for the stream frames) while `available_cwnd` was 500, so the packet is too large for the available cwnd. I am not sure this is a big deal but this behaviour is diffrent from what quiche was doing before.